### PR TITLE
(#103) Updating cactoos version and removing UncheckedScalar

### PR DESCRIPTION
**Pull request by scristalli**
* Updating cactoos to 0.41
* Replacing UncheckedScalar with Unchecked
* Replacing JoinedText with Joined
* Replacing RepeatedFunc with Repeated
* Editing test refuteIfErrorDoesNotMatch to fit new code (now IllegalStateException is wrapped in RuntimeException)

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#125](https://github.com/llorllale/cactoos-matchers/issues/125) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#125](https://github.com/llorllale/cactoos-matchers/issues/125) is assigned to @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov), here is [why](https://www.0crat.com/footprint/C63314D6Z/99dfb83d-024c-41f1-9516-33d63e20357a); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@ilyakharlamov Any way to get a review for this soon? it's blocking a few tasks in cactoos :)

File: src/main/java/org/llorllale/cactoos/matchers/HasLines.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 import org.cactoos.collection.Mapped;
 import org.cactoos.iterable.IterableOf;
 import org.cactoos.list.ListOf;
-import org.cactoos.scalar.UncheckedScalar;
+import org.cactoos.scalar.Unchecked;
 
 /**
  * Allows to check that text has lines considering platform-dependent line
```
Comment (ilyakharlamov) [inline src/main/java/org/llorllale/cactoos/matchers/HasLines.java]: Ctor `HasLines(final Text... lns)` needs to be unit-tested

File: src/main/java/org/llorllale/cactoos/matchers/RunsInThreads.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
 import java.util.concurrent.Future;
```
Comment (ilyakharlamov) [inline src/main/java/org/llorllale/cactoos/matchers/RunsInThreads.java]: Ctor `RunsInThreads()` needs to be unit-tested

File: src/main/java/org/llorllale/cactoos/matchers/RunsInThreads.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
...
 import org.cactoos.Func;
```
Comment (ilyakharlamov) [inline src/main/java/org/llorllale/cactoos/matchers/RunsInThreads.java]: `RunsInThreads(final T object)` needs to be unit-tested

File: src/main/java/org/llorllale/cactoos/matchers/ScalarHasValue.java [hunk 4]
```diff
@@ -82,6 +82,6 @@ protected boolean matchesSafely(final Scalar<T> item) {
     protected void describeMismatchSafely(final Scalar<T> item,
```
Comment (ilyakharlamov) [inline src/main/java/org/llorllale/cactoos/matchers/ScalarHasValue.java]: What so-called `refused bequest` here, `describeMismatchSafely` needs to call super.

**Comment by ilyakharlamov** [comment]
@rultor merge

**Comment by 0crat** [comment]
This pull request [#125](https://github.com/llorllale/cactoos-matchers/issues/125) is assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), here is [why](https://www.0crat.com/footprint/C63314D6Z/7a49be41-8cad-4cf9-bf64-5145dbfa5785); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/llorllale/cactoos/matchers/AssertionTest.java [hunk 5]
```diff
@@ -89,7 +90,9 @@ public void refuteIfResultDoesNotMatch() throws IOException {
      */
     @Test
     public void refuteIfErrorDoesNotMatch() {
-        this.exception.expect(IllegalStateException.class);
+        this.exception.expectCause(
```
Comment (llorllale) [inline src/test/java/org/llorllale/cactoos/matchers/AssertionTest.java]: @scristalli what's the reason for this change?

Comment (scristalli) [reply src/test/java/org/llorllale/cactoos/matchers/AssertionTest.java]: @llorllale due to the wrapping in `IoChecked` done by `Unchecked`:
```
return (new IoChecked(this.origin)).value();
```

as opposed to the old `UncheckedScalar`,
```
return this.origin.value();
```

now the `IllegalStateException` exception raised in the test is wrapped inside a `RuntimeException`, because it is now thrown by `UncheckedText`:
```
throw new RuntimeException(error);
```

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/llorllale/cactoos-matchers/issues/125#issuecomment-498014997))

@g4s8 This is what I know about this job in [C63314D6Z](https://www.0crat.com/p/C63314D6Z), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#125](https://github.com/llorllale/cactoos-matchers/issues/125) is [in scope](http://www.zerocracy.com/policy.html#14) for 9days
  * The role is `REV`
  * The job is assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas) for [2days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @scristalli[/z](https://www.0crat.com/u/scristalli): This user reported the ticket
    * @ilyakharlamov[/z](https://www.0crat.com/u/ilyakharlamov): User was resigned from the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Allorllale%2Fcactoos-matchers%23125%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@vzurauskas @llorllale any way to get this reviewed and merged? It's still blocking a bunch of other issues :)

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#125](https://github.com/llorllale/cactoos-matchers/issues/125) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by llorllale** [comment]
@rultor release, tag is `0.17`
